account_analytic_policy/model/account_move_line.py

```diff
@@ -0,0 +1,72 @@
+# coding: utf-8
+############################################################################
+#    Module Writen For Odoo, Open Source Management Solution
+#
+#    Copyright (c) 2010 Vauxoo - http://www.vauxoo.com
+#    All Rights Reserved.
+#    info Vauxoo (info@vauxoo.com)
+#    coded by: nhomar <nhomar@vauxoo.com>
+############################################################################
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAccount(models.Model):
+    _inherit = "account.account"
+
+    analytic_policy = fields.Selection(
+        [('optional', 'Optional'), ('always', 'Always'), ('never', 'Never')],
+        default='optional',
+        help='"Optional": , the accountant is free to put an analytic account '
+             'on an account move line;\n'
+             '"Always": the accountant will get an error message if there is '
+             'no analytic account;\n'
+             '"Never": the accountant will get an error message if an '
+             'analytic account is present.'
+    )
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.multi
+    def _is_policy_ok(self, vals):
+        account = ('account_id' in vals) and self.env[
+            'account.account'].browse(vals['account_id'])
+        analytic = ('analytic_account_id' in vals) and self.env[
+            'account.account'].browse(vals['analytic_account_id'])
+        if not account:
+            # When write I will not have the account on vals if the account
+            # is actually not modified
+            account = self.account_id
+        policy = account and account.analytic_policy
+        if not policy or not account:
```

Comment:
`or not account`
Is it a valid assertion?

You will not reach this state as account_id is a required field on Journal Items.
https://github.com/odoo/odoo/blob/11.0/addons/account/models/account_move.py#L418

Reply:
It is checked because this is a create before the check on the required part, odoo does on that way in account.